test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Internal/ResponseCacheFilterApplicationModelProviderTest.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Internal
+{
+    public class ResponseCacheFilterApplicationModelProviderTest
+    {
+        [Fact]
+        public void OnProvidersExecuting_DoesNothingIfHandlerHasNoResponseCacheAttributes()
+        {
+            // Arrange
+            var options = new TestOptionsManager<MvcOptions>();
+            var provider = new ResponseCacheFilterApplicationModelProvider(options);
+            var typeInfo = typeof(PageWithoutResponseCache).GetTypeInfo();
+            var context = GetApplicationProviderContext(typeInfo);
+
+            // Act
+            provider.OnProvidersExecuting(context);
+
+            // Assert
+            Assert.Empty(context.PageApplicationModel.Filters);
+        }
+
+        private class PageWithoutResponseCache : Page
+        {
+            public ModelWithoutResponseCache Model => null;
+
+            public override Task ExecuteAsync() => throw new NotImplementedException();
+        }
+
+        [Authorize]
+        public class ModelWithoutResponseCache : PageModel
+        {
+            public void OnGet()
+            {
+            }
+        }
+
+        [Fact]
+        public void OnProvidersExecuting_AddsResponseCacheFilters()
+        {
+            // Arrange
+            var options = new TestOptionsManager<MvcOptions>();
+            var provider = new ResponseCacheFilterApplicationModelProvider(options);
+            var typeInfo = typeof(PageWithResponseCache).GetTypeInfo();
+            var context = GetApplicationProviderContext(typeInfo);
+
+            // Act
+            provider.OnProvidersExecuting(context);
+
+            // Assert
+            Assert.Collection(
+                context.PageApplicationModel.Filters,
+                f => { },
```

Comment:
This is kinda annoying. `ResponseCacheAttribute` is a `IFilterFactory` and ends up adding `ResponseCacheFilter : IActionFilter` to the model. It ultimately does nothing, but it would be nice if we could avoid having it in the list at all.